src/targets/gpu/kernels/include/migraphx/kernels/scatternd.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef MIGRAPHX_GUARD_KERNELS_SCATTERND_HPP
+#define MIGRAPHX_GUARD_KERNELS_SCATTERND_HPP
+
+#include <migraphx/kernels/index.hpp>
+#include <migraphx/kernels/array.hpp>
+#include <migraphx/kernels/dfor.hpp>
+#include <migraphx/kernels/basic_ops.hpp>
+#include <args.hpp>
+
+namespace migraphx {
+
+template <class T1, class T2>
+struct scatternd_settings
+{
+    T1 is_add{};
+    T2 is_mul{};
+};
+
+template <class... Ts>
+constexpr scatternd_settings<Ts...> make_scatternd_settings(Ts... xs)
+{
+    return {xs...};
+}
+
+template <class T, class U, class V, class W, class Settings>
+__device__ void
+scatternd(const T& data_t, const U& indices_t, const V& updates_t, const W& output_t, Settings s)
+{
+    auto index           = make_index();
+    auto i               = index.global;
+    auto updates_shape   = updates_t.get_shape();
+    auto output_shape    = output_t.get_shape();
+    const auto* data_ptr = data_t.data();
+    auto* output_ptr     = output_t.data();
+
+    if(i < output_shape.elements())
+        output_ptr[i] = data_ptr[i];
+    __syncthreads();
```

Comment:
Is this really needed?

Reply:
The __syncthreads() here is not enough since the updates to an element can happen in another workgroup. In this case, if the copy from data happens after the update, then the output is incorrect.  I would suggest use a separate kernel for the copy from data to output since the output elements can be much larger than the update elements, in that case, a lot of workitems do nothing in the kernel.